### PR TITLE
Update amp-cache-guidelines.md with adding a new cache to the AMP ecosystem

### PR DESCRIPTION
**Pull request by pietergreyling**

### CODE REVIEW

File: spec/amp-cache-guidelines.md [hunk 1]
```diff
@@ -88,6 +88,13 @@ Accepted MIME types for fonts include the following prefixes:
 |  binary/octet-stream   |
 |  text/plain (not recommended)   |
 
+## Guidelines: Adding a new cache to the AMP ecosystem
+
+* Contact relevant publishers to make sure they update their CORS policies to include your origin.
+* Publicly announce your cache so people know where to find your documentation.
+* Examples of third-party AMP Cache providers:
```
Comment (cramforce) [inline spec/amp-cache-guidelines.md]: Google is also a third party.

File: spec/amp-cache-guidelines.md [hunk 2]
```diff
@@ -88,6 +88,13 @@ Accepted MIME types for fonts include the following prefixes:
 |  binary/octet-stream   |
 |  text/plain (not recommended)   |
 
+## Guidelines: Adding a new cache to the AMP ecosystem
+
+* Contact relevant publishers to make sure they update their CORS policies to include your origin.
```
Comment (cramforce) [inline spec/amp-cache-guidelines.md]: I know that this is the only way right now, but it obviously doesn't scale, which means UX will be bad :(

File: spec/amp-cache-guidelines.md (hunk 1 above)
Comment (cramforce) [inline spec/amp-cache-guidelines.md]: Lets add Google and remove the third party since there is no first party.

**Comment by pietergreyling** [comment]
Done (like this):
- Examples of AMP Cache providers:
  - [Google AMP Cache](https://developers.google.com/amp/cache/)
  - [Cloudflare AMP Cache](https://www.cloudflare.com/website-optimization/accelerated-mobile-links/)

File: spec/amp-cache-guidelines.md [hunk 3]
```diff
@@ -88,6 +88,14 @@ Accepted MIME types for fonts include the following prefixes:
...
```
Comment (lannka) [inline spec/amp-cache-guidelines.md]: can you also add a link to our [CORS guidelines](https://github.com/ampproject/amphtml/blob/master/spec/amp-cors-requests.md#cors-security-in-amp)?

Comment (pietergreyling) [inline spec/amp-cache-guidelines.md]: @lannka Done as per:
* Contact relevant publishers to make sure they update their CORS policies to include your origin.
  * For further information please refer to the [CORS Security in AMP Guidelines](https://github.com/ampproject/amphtml/blob/master/spec/amp-cors-requests.md#cors-security-in-amp).

**Comment by pietergreyling** [comment]
@lannka @cramforce Are we good on the latest changes?

**Pull request merged**